engine/consensus/sealing/core.go

```diff
@@ -135,8 +135,8 @@ func NewCore(
 // 2) traverse forward all unfinalized(pending) blocks starting from last finalized block.
 // For each block that is being traversed we will collect execution results and process them using sealing.Core.
 func (c *Core) RepopulateAssignmentCollectorTree(payloads storage.Payloads) error {
-	finalizedSnapshot := c.state.Final()
-	finalized, err := finalizedSnapshot.Head()
+	finalizedHeader := c.state.Final()
```

Comment:
is this an intentional change? state.Final() is still the snapshot

Reply:
Right, I'll rename that